src/components/RegisterBox/RegisterBox.js

```diff
@@ -8,34 +8,47 @@ import { connect } from 'react-redux';
 import './RegisterBox.css';
 
 class RegisterBox extends PureComponent {
-
   state = {
     username: '',
     email: '',
     password: '',
     password2: '',
     errors: {},
-  }
+  };
 
   handleRegister = e => {
+    const {
+      username,
+      email,
+      password,
+      password2,
+    } = this.state;
+
     e.preventDefault();
 
     const user = {
-      username: this.state.username,
-      email: this.state.email,
-      password: this.state.password,
-      password2: this.state.password2
-    }
+      username,
+      email,
+      password,
+      password2,
+    };
```

Comment:
Looking again, if your user is everything but `errors` in state, you can just
```
const { errors, ...user } = this.props;
registerUser(user);
```

Reply:
that's cool.